UglyTicTacToe/UglyTicTacToe/Game.aspx.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+namespace UglyTicTacToe
+{
+    public enum Player
+    {
+        None,
+        X,
+        O
+    }
+    public partial class Game : Base
+    {
+       
+
+        private static string TurnKey = "turn";
+        public bool Turn
+        {
+            get
+            {
+                if (ViewState[TurnKey] == null)
+                {
+                    return true;
+                }
+                else
+                {
+                    return (bool)ViewState[TurnKey];
+                }
+            }
+            set
+            {
+                ViewState[TurnKey] = value;
+            }
+        }
+      
+        protected void Page_Load(object sender, EventArgs e)
+        {
+
+        }
+
+        protected void TakeTurn(Button button, int row, int col)
+        {
+            Board.NextTurn();
+            if (Turn == true)
+            {
+                button.Text = "X";
+                Board.GetBoard()[row, col] = Player.X;
+                Turn = false;
+
+                if (Win(Player.X))
+                {
+                    Winner.Text = "Congratulations Player X, You Won!";
+                    return;
+                }
+            }
+            else
+            {
+                button.Text = "O";
+                Turn = true;
+                Board.GetBoard()[row, col] = Player.O;
+                if (Win(Player.O))
+                {
+                    Winner.Text = "Congratulations Player O, You Won!";
+                }
+            }
+            button.Enabled = false;
+        }
+        protected void aa_Click(object sender, EventArgs e)
+        {
+            TakeTurn(aa, 0, 0);
+        }
+
+        protected void ab_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ab, 0, 1);
+        }
+
+        protected void ac_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ac, 0, 2);
+        }
+
+        protected void ba_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ba, 1, 0);
+        }
+
+        protected void bb_Click(object sender, EventArgs e)
+        {
+            TakeTurn(bb, 1, 1);
+        }
+
+        protected void bc_Click(object sender, EventArgs e)
+        {
+            TakeTurn(bc, 1, 2);
+        }
+
+        protected void ca_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ca, 2, 0);
+        }
+
+        protected void cb_Click(object sender, EventArgs e)
+        {
+            TakeTurn(cb, 2, 1);
+        }
+
+        protected void cc_Click(object sender, EventArgs e)
+        {
+            TakeTurn(cc, 2, 2);
+        }
+
+        protected bool Win(Player player)
```

Comment:
You should be able to push this method into your `Base` class so it's not duplicated across your pages.

Reply:
To do that, you'd need to have an abstract `EnableButtons` method in your `Base` class too. We haven't talked about abstract methods yet, but we'll get there soon.